contracts/kernel/KernelConstants.sol

```diff
@@ -5,19 +5,21 @@
 pragma solidity ^0.4.24;
 
 
-contract KernelConstants {
-    /*
-    bytes32 public constant CORE_NAMESPACE = keccak256("core");
-    bytes32 public constant APP_BASES_NAMESPACE = keccak256("base");
-    bytes32 public constant APP_ADDR_NAMESPACE = keccak256("app");
+contract KernelNamespaceConstants {
+    /* Hardcoded constants to save gas
+    bytes32 internal constant KERNEL_CORE_NAMESPACE = keccak256("core");
+    bytes32 internal constant KERNEL_APP_BASES_NAMESPACE = keccak256("base");
+    bytes32 internal constant KERNEL_APP_ADDR_NAMESPACE = keccak256("app");
+    */
+    bytes32 internal constant KERNEL_CORE_NAMESPACE = 0xc681a85306374a5ab27f0bbc385296a54bcd314a1948b6cf61c4ea1bc44bb9f8;
+    bytes32 internal constant KERNEL_APP_BASES_NAMESPACE = 0xf1f3eb40f5bc1ad1344716ced8b8a0431d840b5783aea1fd01786bc26f35ac0f;
+    bytes32 internal constant KERNEL_APP_ADDR_NAMESPACE = 0xd6f028ca0e8edb4a8c9757ca4fdccab25fa1e0317da1188108f7d2dee14902fb;
+}
 
+
+contract KernelConstants {
```

Comment:
Do we need to keep this just for 1 constant? That constant is only used in `KernelProxy`, right?

Reply:
I've now moved all the kernel's default app ids into `KernelDefaultAppIds`.